lib/elixir/lib/io/ansi/docs.ex

```diff
@@ -53,6 +55,37 @@ defmodule IO.ANSI.Docs do
     newline_after_block()
   end
 
+  @doc """
+  Prints documentation metadata (only `since` and `deprecated` for now).
+
+  See `default_options/0` for docs on the supported options.
+  """
+  @spec print_metadata(map, keyword) :: :ok
+  def print_metadata(metadata, options \\ []) when is_map(metadata) do
+    options = Keyword.merge(default_options(), options)
+    print_each_metadata(metadata, options) && newline_after_block()
+  end
+
+  defp print_each_metadata(metadata, options) do
+    Enum.reduce(metadata, false, &print_single_metadata({&1, options}, &2))
+  end
+
+  defp print_single_metadata({{:deprecated, value}, options}, _printed) do
+    [metadata_label(:deprecated, options) | String.split(value, @spaces)]
+    |> write_with_wrap(options[:width], "            ", true)
```

Comment:
Instead of `"            "`, can we compute this value? Is it `String.duplicate(" ", String.length("deprecated") + 2)`? I would prefer to have those things dynamically computed on the key, similar to what @fertapric suggested, than hardcode all individual entries just because adding new binaries metadata in the future (likely to be the most common ones) will be easier and have less code duplication.

Reply:
Fair enough, in this case I'm inclined to refactor and go with something similar to what @fertapric suggested.